extensions/wikia/MarkWikiAsClosed/controllers/MarkWikiAsClosedController.class.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+use Wikia\Logger\Loggable;
+
+class MarkWikiAsClosedController extends WikiaController {
+
+	use Loggable;
+
+	const WIKI_ID = 'wikiId';
+	const REASON = 'reason';
+
+	public function init() {
+		$this->assertCanAccessController();
+	}
+
+	public function markWikiAsClosed() {
+		$context = $this->getContext();
+		$request = $context->getRequest();
+
+		$wikiId = $request->getVal( self::WIKI_ID );
+		$reason = $request->getVal( self::REASON );
+
+		if ( is_null( $wikiId ) || is_null( $reason ) ) {
```

Comment:
A more defensive check `( ! is_numeric( $wikiId ) || empty( $reason ) )` improves error handling.  Consider a malformed request `?controller=MarkWikiAsClosed&method=markWikiAsClosed&wikiId=&reason=`. In this case `$wikiId` and `$reason` won't be `null` and will make `WikiFactory::setPublicStatus()` crash, as it does not have any validation.